### PR TITLE
GHA implementation

### PR DESCRIPTION
**Pull request by ignasirv**
- Implement github actions to run all test for every PR to main/develop
- Fix identity precompiled

### CODE REVIEW

File: .github/workflows/main.yaml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        node-version: [14.x]
+
+    steps:
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v1
+        with:
+          node-version: ${{ matrix.node-version }}
+      - name: Checkout code
+        uses: actions/checkout@v2
+      - name: Clone zkevm-rom
```
Comment (krlosMata) [inline .github/workflows/main.yaml:27]: I think this is not needed since it is done in the previous step

File: .github/workflows/main.yaml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        uses: GuillaumeFalourd/clone-github-repo-action@v2
+        with:
+          owner: '0xPolygonHermez'
+          repository: 'zkevm-rom'
+      - name: setup zkevm-rom
+        run: |
+          cd zkevm-rom          
+          npm i
+          npm run build
+      - name: Clone zkevm-testvectors
```
Comment (krlosMata) [inline .github/workflows/main.yaml:37]: Which version it is downloaded? which branch?
Could we configure it ?

**Review by krlosMata — CHANGES_REQUESTED** [review]

File: .github/workflows/main.yaml (hunk 2 above)
Comment (ignasirv) [reply .github/workflows/main.yaml:37]: Now last commit of main is downloaded, is a git clone. We can configure the branch with a checkout at the run statement

Comment (krlosMata) [reply .github/workflows/main.yaml:37]: Maybe we could set `branch name` as a environment variable to checkout in GHA script

File: main/precompiled/identity.zkasm [hunk 3]
```diff
@@ -1,29 +1,103 @@
...
+    1024 + A + 1 => SP
+    32 - D => D
+    $ => A          :MLOAD(SP)
+                    :CALL(SHRarith)
+    A + C           :MSTORE(bytesToStore)
+                    :CALL(MSTORE32)
+    $ => C          :MLOAD(tmpVarC3)
+    C - 32 => C
+    B + 32 => B
+                    :JMP(IDENTITYinit)
```
Comment (invocamanman) [inline main/precompiled/identity.zkasm:46]: Could be optimized since B is always 0

File: main/precompiled/identity.zkasm [hunk 4]
```diff
@@ -1,29 +1,103 @@
...
+IDENTITYfinal:
+    ${B%32} => D
+    1024 + ${B/32} => SP
+    $ => A          :MLOAD(SP)
+                    :CALL(SHLarith)
+    1024 + ${B/32} + 1 => SP
+    C - 32 + D => D
+    D               :JMPN(IDENTITYxor)
```
Comment (invocamanman) [inline main/precompiled/identity.zkasm:54]: This code in this case in unreachable bc ${B%32} => D will be always 0
But in the case of the calldata i think the case where C + D is 32 should also go to the same case escenario

File: main/precompiled/identity.zkasm [hunk 5]
```diff
@@ -1,29 +1,103 @@
...
+    A => B
+    $ => A          :MLOAD(SP)
+    32 - D => D
+                    :CALL(SHRarith)
+    32 - C => D
+                    :CALL(SHLarith)
```
Comment (invocamanman) [inline main/precompiled/identity.zkasm:60]: I think both cases shoudl be 32 - C => D

File: main/precompiled/identity.zkasm [hunk 6]
```diff
@@ -1,29 +1,103 @@
...
+IDENTITYfinal:
+    ${B%32} => D
+    1024 + ${B/32} => SP
+    $ => A          :MLOAD(SP)
+                    :CALL(SHLarith)
+    1024 + ${B/32} + 1 => SP
+    C - 32 + D => D
+    D               :JMPN(IDENTITYxor)
+    A => B
+    $ => A          :MLOAD(SP)
```
Comment (invocamanman) [inline main/precompiled/identity.zkasm:56]: I think it should be SP + 1

Comment (ignasirv) [reply main/precompiled/identity.zkasm:38]: I will refactor all this part

File: main/precompiled/identity.zkasm [hunk 7]
```diff
@@ -1,36 +1,75 @@
...
+    ; Copy from Stack to memory
+    C - 1           :JMPN(IDENTITYreturn)
     C - 32          :JMPN(IDENTITYfinal)
+    1024 + A => SP
+    $ => B          :MLOAD(SP)
+    B               :MSTORE(bytesToStore)
+                    :CALL(MSTORE32)
+    A + 1 => A
+    C - 32 => C
+    B + 32 => B
```
Comment (invocamanman) [inline main/precompiled/identity.zkasm:34]: si not necessary to increment B here

File: main/precompiled/identity.zkasm [hunk 8]
```diff
@@ -1,36 +1,75 @@
...
+    $ => A          :MLOAD(SP)
+    32 - C => D
+                    :CALL(SHRarith)
+                    :CALL(SHLarith)
+    A               :MSTORE(bytesToStore)
+                    :CALL(MSTOREX)
+IDENTITYreturn:
+    ; Copy from memory current CTX to memory origin CTX
+    0 => E
+    $ => C          :MLOAD(retCallLength)
```
Comment (invocamanman) [inline main/precompiled/identity.zkasm:47]: we should check the retCallLen and the argsLengthCall and copy the minimum of those 2

**Review by invocamanman — APPROVED** [review]

**Review by invocamanman — APPROVED** [review]

**Pull request merged**